Objects/obmalloc.c

```diff
@@ -1231,6 +1231,21 @@ _PyObject_XDecRefDelayed(PyObject *ptr)
 }
 #endif
 
+#ifdef Py_GIL_DISABLED
+void
+_PyObject_XSetRefDelayed(PyObject **ptr, PyObject *value)
+{
+    PyObject *old = FT_ATOMIC_LOAD_PTR_ACQUIRE(*ptr);
```

Comment:
We shouldn't need atomics to *load* the pointer within the lock since nobody else can safely modify it concurrently. (We still need the atomics to *write* the ptr).

```suggestion
    PyObject *old = *ptr;
```